release notes/v0.39.0.md

```diff
@@ -0,0 +1,60 @@
+
+Over the last several k6 releases, we have also slowly been refactoring and improving the metrics internals in k6 (see [#2071](https://github.com/grafana/k6/pull/2071), [#2330](https://github.com/grafana/k6/issues/2330), [#2426](https://github.com/grafana/k6/pull/2426), [#2463](https://github.com/grafana/k6/pull/2463), [#2433](https://github.com/grafana/k6/pull/2433), [#2442](https://github.com/grafana/k6/pull/2442), among others). This has already brought many side-benefits and minor bugfixes, and we still have a lot of work left (e.g. [#1889](https://github.com/grafana/k6/issues/1889), [#572](https://github.com/grafana/k6/issues/572), [#2430](https://github.com/grafana/k6/issues/2430), [#1831](https://github.com/grafana/k6/issues/1831)), but we have finally reached the point where we can almost start implementing some major new features effectively!
+
+One of the upcoming next big steps is the introduction of a "time series" concept internally in k6 ([#2580](https://github.com/grafana/k6/issues/2580)). We'll start to efficiently group samples (i.e. metric measurements) with the same metric and tags into a single `TimeSeries`, which would unlock many of possibilities that were previously too inefficient to implement. First among them will probably be the refactoring and improvement of the [Prometheus remote-write output extension](https://github.com/grafana/xk6-output-prometheus-remote/) and its eventual merging into k6 as a built-in k6 output.
+
+Another upcoming metrics-related change would be the refactoring of [metric tags](https://k6.io/docs/using-k6/tags-and-groups/#tags) into two distinct types - ones that can be used for indexing (so, in the `TimeSeries` mentioned above, in [sub-metric thresholds](https://k6.io/docs/using-k6/thresholds/#thresholds-on-tags), in certain outputs that do aggregation based on tags, etc. ) and ones that cannot (e.g. high cardinality metadata data that's often unique for every data point). The exact new APIs are still being discussed, but we'll try to make them maximally backwards-compatible. The change is necessary both to implement the time series refactoring mentioned above and also for things like tracing support ([#2128](https://github.com/grafana/k6/issues/2128)).
+
+We'll probably need to make at least a few minor breaking changes here. We'll try to give at least one k6 version's notice whenever possible. That is, we'll try to print a warning when a k6 behavior that will be broken in the next k6 release is used. The biggest potentially breaking change is that the current [`url`, `error`, `vu` and `iter` system tags](https://k6.io/docs/using-k6/tags-and-groups/#system-tags) will be made non-indexable, to reflect their usually very high cardinality, so they won't be usable in thresholds. Instead of `error`, `error_code` should be used there. And instead of `url`, users should already use `name`, see the [documentation about URL grouping](https://k6.io/docs/using-k6/http-requests/#url-grouping).
```

Comment:
```suggestion
We'll probably need to make at least a few minor breaking changes here. We'll try to give at least one k6 version's notice whenever possible. That is, we'll try to print a warning when a k6 behavior that will be broken in the next k6 release is used.

:warning: The biggest potentially breaking change is that the current [`url`, `error`, `vu` and `iter` system tags](https://k6.io/docs/using-k6/tags-and-groups/#system-tags) will be made non-indexable, to reflect their usually very high cardinality, so they won't be usable in thresholds. [error_code](https://k6.io/docs/javascript-api/error-codes/) should be used as a replacement for `error`. And instead of `url`, users should already use `name`, see the [documentation about URL grouping](https://k6.io/docs/using-k6/http-requests/#url-grouping).
```

Reply:
See latest version, changed it somewhat like this, but without the :warning:.